### DISCUSSION TITLE
libxslt not being able to find installed stylesheets

### DISCUSSION THREAD

**Discussion by jtojnar** [Homebrew/discussions#3122] [Everyday usage]
When running Meson tests on MacOS CI ([this PR](https://github.com/mesonbuild/meson/pull/10186)), `gtk-doc` failed to run because `xsltproc` (from `libxslt`) was not able to found `docbook-xsl` stylesheets, even though they were installed through `brew install gtk-doc`. `libxml2`-based tools like `libxslt` support [XML catalogues](https://en.wikipedia.org/wiki/XML_catalog) as a mechanism for finding XSL stylesheets but, since Brew installs it to an unconventional location (`/usr/local/etc/xml/catalog` instead of the expected `/etc/xml/catalog`), they will not be able to see the installed files.

I know I can set `XML_CATALOG_FILES` but is that really the expected solution?

There was an issue about this but it got closed: https://github.com/Homebrew/homebrew-core/issues/70769

**Comment by SMillerDev** [comment]
> I know I can set XML_CATALOG_FILES but is that really the expected solution?

Yeah, that's the variable to control where to find catalogs

**Reply by jtojnar** [reply]
I know. But should not `libxml2` installed by Homebrew be aware of Homebrew’s XML catalog out of the box?

**Reply by SMillerDev** [reply]
AFAIK `libxml2` is rarely used from Homebrew since it's provided by macOS. It might be an oversight or maybe there is just no way to edit this location at compile time. I don't know.

**Reply by carlocab** [reply]
> I know. But should not `libxml2` installed by Homebrew be aware of Homebrew’s XML catalog out of the box?

PRs to improve this behaviour welcome.

**Reply by jtojnar** [reply]
> It might be an oversight or maybe there is just no way to edit this location at compile time. I don't know.

You can pass `-DXML_XML_DEFAULT_CATALOG` to the compiler to change the default path:

https://gitlab.gnome.org/GNOME/libxml2/-/blob/95766541622b3a66c929e2e272b0252750d0174d/catalog.c#L71

> PRs to improve this behaviour welcome.

Sorry, Homebrew is three degrees of separation from the project I was trying to fix (I do not personally even use MacOS) so I won’t be able to do that.
